### PR TITLE
Move documentation to keycloak-client

### PR DESCRIPTION
**Pull request by rmartinc**
Moving the documentation to the keycloak-client. The idea is the commented chapters are moved to the new keycloak-client project as guides. The plugin used in keycloak is copied but removed the specific keycloak parts (it's simpler). I don't think this will be changed a lot.

In order to complete the issue we need two more PRs in different projects:

* The PR to remove the chapters in keycloak: https://github.com/keycloak/keycloak/pull/32187
* The changes in keycloak-web to add the new section in guides (this one cannot be merged until a real keycloak-client package with the docs is released): https://github.com/keycloak/keycloak-web/pull/482

The new guides are generated in the keycloak-web as a new section:

Screenshot from 2024-08-16 11-13-40

And each section is just a normal guide (for example the first one):

Screenshot from 2024-08-16 11-15-07

Draft for the moment.

### CODE REVIEW

**Comment by mposolda** [comment]
@rmartinc Thanks!

I have one small concern related to policy-enforcer javascript adapter. The JS adapter is still part of the main Keycloak repository and hence makes sense if it is still documented there. So can be better to either keep that one (file [enforcer-js-adapter.adoc`) in the keycloak and either keep in authorization-services docs or move to the JS adapter docs (which was already converted to guides). WDYT?

Copying the maven plugin from Keycloak is not so great, but IMO it is OK. Especially considering that it is relatively simple thing, which does not change too often and I don't see much better solutions for it, which are simple at the same time (anything more fancy can make things unnecessary complicated, so not sure if it worth it...)

**Comment by rmartinc** [comment]
> I have one small concern related to policy-enforcer javascript adapter. The JS adapter is still part of the main Keycloak repository and hence makes sense if it is still documented there. So can be better to either keep that one (file enforcer-js-adapter.adoc) in the keycloak and either keep in authorization-services docs or move to the JS adapter docs (which was already converted to guides). WDYT?

Yes, no problem, I simply moved the whole chapter. :smile: I will move back to the documentation. Then I ned to change two of the three PRs.

**Comment by rmartinc** [comment]
@mposolda The policy-enforcer javascript adapter chapter has been maintained in the authorization services documentation and removed from the client.

File: docs/guides/securing-apps/admin-client.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="{project_name} admin client"
+priority=500
+summary="Using the {project_name} admin client to access the REST API">
+
+The {project_name} admin client is a JAVA library that facilitates the access and usage of the {project_name} Admin REST API. To use it from your application add a dependency on the
+`keycloak-admin-client` library. For example using maven:
```
Comment (stianst) [inline docs/guides/securing-apps/admin-client.adoc:10]:
```suggestion
`keycloak-admin-client` library. For example using Maven:
```

File: docs/guides/securing-apps/admin-client.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="{project_name} admin client"
+priority=500
+summary="Using the {project_name} admin client to access the REST API">
+
+The {project_name} admin client is a JAVA library that facilitates the access and usage of the {project_name} Admin REST API. To use it from your application add a dependency on the
```
Comment (stianst) [inline docs/guides/securing-apps/admin-client.adoc:9]:
```suggestion
The {project_name} admin client is a Java library that facilitates the access and usage of the {project_name} Admin REST API. To use it from your application add a dependency on the
```

File: docs/guides/securing-apps/admin-client.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="{project_name} admin client"
+priority=500
+summary="Using the {project_name} admin client to access the REST API">
```
Comment (stianst) [inline docs/guides/securing-apps/admin-client.adoc:7]:
```suggestion
summary="Using the {project_name} admin client to access the {project_name} Admin REST API">
```

File: docs/guides/attributes.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
+:project_name: Keycloak
+:archivebasename: keycloak client
+:archivebasename: keycloak client
```
Comment (stianst) [inline docs/guides/attributes.adoc:3]:
```suggestion
:archivebasename: keycloak client
```

File: docs/guides/attributes.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+:project_doc_base_url: https://www.keycloak.org/docs/latest
+:apidocs_name: API Documentation
+:apidocs_link: https://www.keycloak.org/docs/latest/api_documentation/
+:authorizationguide_name: Authorization Services Guide
+:authorizationguide_name_short: Authorization Services
+:authorizationguide_link: {project_doc_base_url}/authorization_services/
+:kc_base_path:
```
Comment (stianst) [inline docs/guides/attributes.adoc:10]:
```suggestion
```

Could probably remove `kc_base_path` and just hard-code to `/' in the guides. Seems a bit pointless to have it here.

File: docs/guides/attributes.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+:kc_js_path: /js
```
Comment (stianst) [inline docs/guides/attributes.adoc:11]:
```suggestion
```

Unused, so remove

File: docs/guides/attributes.adoc [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+:kc_realms_path: /realms
```
Comment (stianst) [inline docs/guides/attributes.adoc:12]:
```suggestion
```

Unused, so remove

File: docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideBuilder.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    private final Log log;
+
+    public GuideBuilder(File srcDir, File targetDir, Log log, MavenProject project) throws IOException {
+        this.srcDir = srcDir;
+        this.targetDir = targetDir;
+        this.log = log;
+
+        Map<String, Object> globalAttributes = new HashMap<>();
+        globalAttributes.put("ctx", new Context(srcDir));
+        globalAttributes.put("version", project.getVersion());
```
Comment (stianst) [inline docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideBuilder.java:26]:
```suggestion
        globalAttributes.put("client-version", project.getVersion());
```

File: docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideBuilder.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        globalAttributes.put("properties", project.getProperties());
```
Comment (stianst) [inline docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideBuilder.java:27]:
```suggestion
```

This is not needed as you can get version from the `version` attribute, although that should probably be renamed to `client-version` to make it explicit and not risk ending up using the Keycloak server version when building docs.

**Comment by rmartinc** [comment]
Thanks @stianst! All done! I have picked all the commits and done the manual changes. The final version attribute for the client is `client_version` instead of `client-version` because the hyphen seems to be problematic. Thanks!

**Comment by rmartinc** [comment]
Rebased! It was needed.

**Review by mposolda — APPROVED** [review]

**Review by pskopek — APPROVED** [review]

**Comment by mposolda** [comment]
@rmartinc @stianst @pskopek Thanks for the PR and reviews

**Pull request merged**